src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPTypesTest.java

```diff
@@ -0,0 +1,512 @@
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.tvp;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerCallableStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerDataTable;
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+@RunWith(JUnitPlatform.class)
+public class TVPTypesTest extends AbstractTest {
+
+    private static Connection conn = null;
+    static Statement stmt = null;
+    static ResultSet rs = null;
+    static SQLServerDataTable tvp = null;
+    private static String tvpName = "MaxTypesTVP";
+    private static String charTable = "MaxTypesTVPTable";
+    private static String procedureName = "procedureThatCallsTVP";
+
+    /**
+     * Test a longvarchar support
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testLongVarchar() throws SQLException {
+        createTables("varchar(max)");
+        createTVPS("varchar(max)");
+
+        StringBuffer buffer = new StringBuffer();
+        for (int i = 0; i < 9000; i++)
+            buffer.append("a");
+
+        tvp = new SQLServerDataTable();
+        tvp.addColumnMetadata("c1", java.sql.Types.LONGVARCHAR);
+        tvp.addRow(buffer.toString());
+
+        SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) connection
+                .prepareStatement("INSERT INTO " + charTable + " select * from ? ;");
+        pstmt.setStructured(1, tvpName, tvp);
+
+        pstmt.execute();
+
+        if (null != pstmt) {
+            pstmt.close();
+        }
+    }
+
+    /**
+     * Test longnvarchar
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testLongNVarchar() throws SQLException {
+        createTables("nvarchar(max)");
+        createTVPS("nvarchar(max)");
+
+        StringBuffer buffer = new StringBuffer();
+        for (int i = 0; i < 8001; i++)
+            buffer.append("سس");
+
+        tvp = new SQLServerDataTable();
+        tvp.addColumnMetadata("c1", java.sql.Types.LONGNVARCHAR);
+        tvp.addRow(buffer.toString());
+
+        SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) connection
+                .prepareStatement("INSERT INTO " + charTable + " select * from ? ;");
+        pstmt.setStructured(1, tvpName, tvp);
+
+        pstmt.execute();
+
```

Comment:
It may be better to add verification here (and in testLongVarchar) like you have in testXML